src/app/child-dev-project/children/child-details/form/form.component.ts

```diff
@@ -0,0 +1,144 @@
+import { Component, Input, OnChanges, SimpleChanges } from "@angular/core";
+import { Child } from "../../model/child";
+import { FormBuilder, FormGroup, Validators } from "@angular/forms";
+import { EntityMapperService } from "../../../../core/entity/entity-mapper.service";
+import { AlertService } from "../../../../core/alerts/alert.service";
+import { ChildPhotoService } from "../../child-photo-service/child-photo.service";
+import { Router } from "@angular/router";
+import { SessionService } from "../../../../core/session/session-service/session.service";
+
+@Component({
+  selector: "app-form",
+  templateUrl: "./form.component.html",
+  styleUrls: ["./form.component.scss"],
+})
+export class FormComponent implements OnChanges {
+  @Input() child: Child;
+
+  creatingNew = false;
+  isAdminUser: boolean;
+  enablePhotoUpload = false;
+
+  editing: boolean = false;
+  form: FormGroup;
+  validateForm: boolean = false;
+  config;
+
+  constructor(
+    private fb: FormBuilder,
+    private entityMapperService: EntityMapperService,
+    private alertService: AlertService,
+    private childPhotoService: ChildPhotoService,
+    private router: Router,
+    private sessionService: SessionService
+  ) {
+    this.isAdminUser = this.sessionService.getCurrentUser().admin;
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.hasOwnProperty("child")) {
+      this.initForm();
+    }
+  }
+
+  onInitFromDynamicConfig(config: any) {
+    this.child = config.child;
+    this.config = config.config;
+    this.initForm();
+    if (!config.child.name) {
```

Comment:
that seems slightly hacky ... who knows how field names and required fields change in the future. Maybe we should add an explicit flag `isCreatingNew` to the config object instead?